vnext/Microsoft.ReactNative.Managed.CodeGen.UnitTests/CodeGen/Lkg/Microsoft.ReactNative.Managed.CodeGen.UnitTests.CodeGen.CodeGenStructSerializerTests--AllPropertyTypes.lkg

```diff
@@ -0,0 +1,40 @@
+internal void CreateObjectSerializers()
+{
+    global::Microsoft.ReactNative.Managed.JSValueReaderCodeGen<global::TestClass.MyStruct>.ReadValue = (global::Microsoft.ReactNative.IJSValueReader reader, out global::TestClass.MyStruct value) =>
+    {
+        value = new global::TestClass.MyStruct();
+        if (reader.ValueType == global::Microsoft.ReactNative.JSValueType.Object)
+        {
+            while (reader.GetNextObjectProperty(out string propertyName))
+            {
+                switch (propertyName)
+                {
+                    case "Y":
+                        value.Y = global::Microsoft.ReactNative.Managed.JSValueReader.ReadValue<int>(reader);
+                        break;
+                    case "Z":
+                        value.Z = global::Microsoft.ReactNative.Managed.JSValueReader.ReadValue<int>(reader);
```

Comment:
>global::Microsoft.ReactNative.Managed.JSValueReader.ReadValue<int>(reader); [](start = 34, length = 75)

Can we generate it as an extension method?
reader.ReadValue<int>(). The same is for the writer.

Reply:
We can, but we run the risk of collision with any extension methods or types defined by the user. I purposely chose to generate unambigious code.